### PR TITLE
Create services and render them

### PR DESCRIPTION
**Pull request by tsimvris**

### CODE REVIEW

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	const router = useRouter();
+	const {
+		register,
+		formState: {errors},
+		handleSubmit,
+	} = useForm({
+		criteriaMode: 'all',
+	});
+	const onSubmit = data => {
+		let service = {
```
Comment (ahohnsen) [inline src/components/Forms/CreateNewServiceForm.jsx:25]: Why not use a `const` here? :)
```suggestion
		const service = {
```

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		router.push({
+			pathname: '/services',
+		});
+	};
+
+	return (
+		<StyledWrapper>
+			<StyledForm onSubmit={handleSubmit(onSubmit)}>
+				<StyledLabel>
+					Service Name{' '}
```
Comment (ahohnsen) [inline src/components/Forms/CreateNewServiceForm.jsx:42]:
```suggestion
					Service Name
```

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+						type="number"
+						{...register('servicePricePerHour', {
+							required: true,
+							maxLength: {
+								value: 4,
+								message: 'This input exceed the maximal length.',
+							},
+							minLength: {
+								value: 1,
+								message: 'This input requires atleast 1 number.',
```
Comment (ahohnsen) [inline src/components/Forms/CreateNewServiceForm.jsx:99]:
```suggestion
								message: 'This input requires at least 1 number.',
```

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+						type="number"
+						{...register('taxKey', {
+							required: {value: true, message: 'This is required.'},
+							minLength: {
+								value: 1,
+								message: 'Please enter a valid Tax Key(7 or 19 %).',
+							},
+							maxLength: {
+								value: 2,
+								message: 'This input exceed the maximal length.',
```
Comment (ahohnsen) [inline src/components/Forms/CreateNewServiceForm.jsx:126]:
```suggestion
								message: 'This input exceeds the maximal length.',
```

File: src/pages/CreateNewService.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import CreateNewServiceForm from '../components/Forms/CreateNewServiceForm';
+export default function CreateNewService() {
+	return (
+		<>
+			<CreateNewServiceForm />
+		</>
```
Comment (ahohnsen) [inline src/pages/CreateNewService.jsx:6]: As you are only returning one child you do not need the fragment.

```suggestion
			<CreateNewServiceForm />
```

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+		let service = {
+			id: nanoid(),
+			serviceName: data.serviceName,
+			serviceDescription: data.serviceDescription,
+			servicePricePerHour: data.servicePricePerHour,
+			serviceTaxKey: data.taxKey,
+		};
+		addService(service);
+		router.push({
+			pathname: '/services',
```
Comment (ahohnsen) [inline src/components/Forms/CreateNewServiceForm.jsx:34]: I guess you want to redirect somewhere else here:

```suggestion
			pathname: '/Services',
```

However I would recommend you to stick writing your page files in lowercase e.g. _services_, _create-new-services_ (kebab-case) 🍢.

File: src/pages/Services.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+export default function Services() {
+	const router = useRouter();
+	const services = useServiceStore(state => state.services);
+
+	return (
+		<StyledWrapper>
+			<StyledButton
+				onClick={() => {
+					router.push({
+						pathname: '/CreateNewService',
```
Comment (ahohnsen) [inline src/pages/Services.jsx:18]:
```suggestion
						pathname: '/create-new-service',
```

**Review by ahohnsen — CHANGES_REQUESTED** [review]
Hey Petros,

Nicely done 🚀 Please have a look at my comments.
I think there should be a button on your home/index page to choose between "adding a new client" and "adding a new service". In your preview it looks like only the first choice is available, so the user cannot navigate to the service page.

File: src/components/ClientUI/styledLi.jsx [hunk 8]
```diff
@@ -2,6 +2,7 @@ import styled from 'styled-components';
 
 const StyledLi = styled.li`
```
Comment (JessicaLoers) [inline src/components/ClientUI/styledLi.jsx:3]: Actually there's `no need`  to position your `<li>` `relative`, neither to position your `<button> and <span>`  (see comments) `absolute`.  👉 This will cause weird bugs 

My suggestion is:  

```suggestion
const StyledLi = styled.li`
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding: 0 20px;
	width: 350px;
	height: 50px;
	margin: 10px;
	border-radius: 15px;
	background-color: #fc814a;
`;
```

File: src/components/Forms/StyledComponents/styledEditButton.jsx [hunk 9]
```diff
@@ -1,6 +1,9 @@
 import styled from 'styled-components';
 
 const StyledEditButton = styled.button`
+	position: absolute;
+	top: 5px;
+	right: 5px;
```
Comment (JessicaLoers) [inline src/components/Forms/StyledComponents/styledEditButton.jsx:6]: If you change the styling of the parent `<li> `like I mentioned in my comment, you can remove these lines 🧹 for the `StyledEditButton.js `AND for the `StyledSpan.js` **(which is not part of this PR)**  .

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+		const service = {
+			id: nanoid(),
+			serviceName: data.serviceName,
+			serviceDescription: data.serviceDescription,
+			servicePricePerHour: data.servicePricePerHour,
+			serviceTaxKey: data.taxKey,
+		};
+		addService(service);
+		router.push({
+			pathname: '/services',
```
Comment (JessicaLoers) [inline src/components/Forms/CreateNewServiceForm.jsx:34]: 👍 Now, the path and the and the file name  look familiar.

File: src/pages/clients.jsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+						pathname: '/',
+					});
+				}}
+			>
+				Home
+			</StyledButton>
+			<StyledButton
+				onClick={() => {
+					router.push({
+						pathname: '/create-new-client',
```
Comment (JessicaLoers) [inline src/pages/clients.jsx:32]: 👍 Now, the path and the and the file name look familiar.

File: src/pages/index.jsx [hunk 12]
```diff
@@ -1,51 +1,32 @@
...
 import StyledButton from '../components/styledButton';
-import StyledWrapper from '../components/styledClientWrapper';
-import useClientStore from '../hooks/useClientStore';
-
-export default function ClientsPage() {
-	const clients = useClientStore(state => state.clients);
+export default function Homepage() {
 	const router = useRouter();
-	console.log(clients);
+	const DynamicWrapper = dynamic(() => import('../components/styledClientWrapper'), {
```
Comment (JessicaLoers) [inline src/pages/index.jsx:7]: 👍

**Review by JessicaLoers — COMMENTED** [review]
**Hi @tsimvris**  ✌️
like I told you there're some styling issues you should be aware of, because they can lead you in some mean bugs in the future. 

I noticed that you implement @ahohnsen suggestions  and fixed the hydration error 🚀

**Review by JessicaLoers — APPROVED** [review]
👍

File: src/components/Forms/CreateNewClientForm.jsx [hunk 13]
```diff
@@ -139,7 +144,7 @@ export default function CreateNewClientForm() {
 				<StyledLabel>
 					Tax ID
 					<StyledInput
-						type="number"
+						type="text"
```
Comment (JessicaLoers) [inline src/components/Forms/CreateNewClientForm.jsx:147]: 👍

File: src/components/Forms/CreateNewClientForm.jsx [hunk 14]
```diff
@@ -164,6 +173,7 @@ export default function CreateNewClientForm() {
 					/>
 				</StyledLabel>
 				<StyledSubmitButton type="submit">Submit</StyledSubmitButton>
+				<StyledHinweis>*All input fields are required</StyledHinweis>
```
Comment (JessicaLoers) [inline src/components/Forms/CreateNewClientForm.jsx:176]: Please be consistent and name your components in English: 
`<StyledNotice />`

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 15]
```diff
@@ -79,24 +82,28 @@ export default function CreateNewServiceForm() {
...
+								<StyledError key={type}>{message}</StyledError>
 							))
 						}
 					/>
 				</StyledLabel>
 				<StyledLabel>
 					Price / Hour
 					<StyledInput
-						type="number"
+						type="text"
```
Comment (JessicaLoers) [inline src/components/Forms/CreateNewServiceForm.jsx:93]: 👍

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 16]
```diff
@@ -106,40 +113,36 @@ export default function CreateNewServiceForm() {
...
+							<input
+								name="taxKey"
+								type="radio"
+								value="19%"
+								checked
+								{...register('taxKey')}
+							/>
+							19%
+						</label>
+					</StyledFieldset>
```
Comment (JessicaLoers) [inline src/components/Forms/CreateNewServiceForm.jsx:142]: 👍

File: src/components/Forms/CreateNewServiceForm.jsx [hunk 17]
```diff
@@ -106,40 +113,36 @@ export default function CreateNewServiceForm() {
...
 				</StyledLabel>
-
 				<StyledSubmitButton type="submit">Submit</StyledSubmitButton>
+				<StyledHinweis>*All input fields are required</StyledHinweis>
```
Comment (JessicaLoers) [inline src/components/Forms/CreateNewServiceForm.jsx:145]: Same as above 😉

**Review by JessicaLoers — APPROVED** [review]
well done 👍
